ocs_ci/ocs/cluster.py

```diff
@@ -1209,6 +1209,37 @@ def check_osd_tree_1az_aws(osd_tree, number_of_osds):
     return check_osds_in_hosts_osd_tree(all_hosts_flatten, osd_tree)
 
 
+def check_ceph_osd_tree():
+    """
+    Checks whether an OSD tree is created/modified correctly.
+    It is a summary of the previous functions: 'check_osd_tree_1az_vmware',
+    'check_osd_tree_3az_aws', 'check_osd_tree_1az_aws'.
+
+    Returns:
+         Boolean: True, if the ceph osd tree is formed correctly. Else False
+
+    """
+    osd_pods = pod.get_osd_pods()
+    # 'ceph osd tree' should show the new osds under right nodes/hosts
+    #  Verification is different for 3 AZ and 1 AZ configs
+    ct_pod = pod.get_ceph_tools_pod()
+    tree_output = ct_pod.exec_ceph_cmd(ceph_cmd='ceph osd tree')
+    if config.ENV_DATA['platform'].lower() == constants.VSPHERE_PLATFORM:
```

Comment:
what about bare metal?

Reply:
Should we automate the process of "node replacement" also for bare metal? currently, it's just for AWS and vSphere.
Or just this specific function 'check_ceph_osd_tree' we need to expand for bare metal?

Reply:
Or just this specific function 'check_ceph_osd_tree' we need to expand for bare metal? -> Yes, let's do that